test/boost/mutation_test.cc

```diff
@@ -703,68 +700,85 @@ SEASTAR_TEST_CASE(test_cell_ordering) {
         BOOST_REQUIRE(compare_atomic_cell_for_merge(c2, c1) == 0);
     };
 
+    testlog.debug("Live cells with same value are equal");
     assert_equal(
         atomic_cell::make_live(*bytes_type, 0, bytes("value")),
         atomic_cell::make_live(*bytes_type, 0, bytes("value")));
 
+    testlog.debug("Live cells are ordered before expiring cells");
     assert_order(
         atomic_cell::make_live(*bytes_type, 1, bytes("value")),
         atomic_cell::make_live(*bytes_type, 1, bytes("value"), expiry_1, ttl_1));
 
+    testlog.debug("Live cells are ordered before expiring cells, regardless of their value");
+    assert_order(
+        atomic_cell::make_live(*bytes_type, 1, bytes("value2")),
+        atomic_cell::make_live(*bytes_type, 1, bytes("value1"), expiry_1, ttl_1));
```

Comment:
Only expiring/dead cells have a write time. Live cells don't have one.

Reply:
I see. It kinda makes sense to me now too: they want to prefer the value that has the longest "expire_at" and the alive cell has an infinite value for this (formally).

Reply:
According to CASSANDRA-14592 they keep expiring cells because they'll become tombstones in the future and then they'll trump live cells.  If they kept the non-expiring cell (as it will survive longer), the expiration would be lost.  Their description of the issue indicates that this inconsistency depends on WHEN the comparison took place, as if it was made early enough, before the cell expired, then the live cell could have been picked (based on its value), but if the comparison took place later on, after the cell expired, then it would be considered as a tombstone and then it would prevail, so preferring expiring cells over live ones is there to be consistent with the rule the prefers tombstones over live cells.